Flaky test in `package service`

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa0 pc=0x661bf6]

goroutine 137 [running]:
panic(0x9cb740, 0xc42000c0b0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
testing.tRunner.func1(0xc4203db5c0)
	/usr/local/go/src/testing/testing.go:579 +0x25d
panic(0x9cb740, 0xc42000c0b0)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/kolide/kolide-ose/vendor/github.com/stretchr/testify/assert.isNumericType(0x0, 0x0, 0x0)
	/go/src/github.com/kolide/kolide-ose/vendor/github.com/stretchr/testify/assert/assertions.go:310 +0x26
github.com/kolide/kolide-ose/vendor/github.com/stretchr/testify/assert.formatUnequalValues(0x0, 0x0, 0x9baac0, 0xc420122460, 0x0, 0x0, 0xc4202442a0, 0xc420122450)
	/go/src/github.com/kolide/kolide-ose/vendor/github.com/stretchr/testify/assert/assertions.go:300 +0xbb
github.com/kolide/kolide-ose/vendor/github.com/stretchr/testify/assert.Equal(0xd23960, 0xc4203dab40, 0x0, 0x0, 0x9baac0, 0xc420122460, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/kolide/kolide-ose/vendor/github.com/stretchr/testify/assert/assertions.go:281 +0xf8
github.com/kolide/kolide-ose/server/service.TestRequestPasswordReset.func3(0xc4203db5c0)
	/go/src/github.com/kolide/kolide-ose/server/service/service_users_test.go:99 +0x2b1
testing.tRunner(0xc4203db5c0, 0xc4203f80c0)
	/usr/local/go/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:646 +0x2ec
FAIL	github.com/kolide/kolide-ose/server/service	2.639s
?   	github.com/kolide/kolide-ose/server/version	[no test files]
?   	github.com/kolide/kolide-ose	[no test files]
Makefile:103: recipe for target 'test-go' failed
make: *** [test-go] Error 1

```

I can't seem to get it to fail locally, or even on CI, but it looks like it happens due to the mix of `t` and `st` values of `testing.T` inside a `Parallel()` t.Run(). 

Either remove Parallel execution or keep `st` consistent inside `t.Run`.

--- a/server/service/service_users_test.go
+++ b/server/service/service_users_test.go
@@ -2,7 +2,6 @@ package service
 
 import (
 	"errors"
-	"strconv"
 	"testing"
 	"time"
 
@@ -86,9 +85,9 @@ func TestRequestPasswordReset(t *testing.T) {
 	}
 
 	for _, tt := range requestPasswordResetTests {
-		tt := tt
-		t.Run("", func(st *testing.T) {
-			st.Parallel()
+		t.Run("", func(t *testing.T) {
+			t.Parallel()
+			tt := tt
 			ctx := context.Background()
 			if tt.vc != nil {
 				ctx = viewer.NewContext(ctx, *tt.vc)
@@ -98,14 +97,14 @@ func TestRequestPasswordReset(t *testing.T) {
 			serviceErr := svc.RequestPasswordReset(ctx, tt.email)
 			assert.Equal(t, tt.wantErr, serviceErr)
 			if tt.vc != nil && tt.vc.IsAdmin() {
-				assert.False(st, mailer.Invoked, "email should not be sent if reset requested by admin")
-				assert.True(st, tt.user.AdminForcedPasswordReset, "AdminForcedPasswordReset should be true if reset requested by admin")
+				assert.False(t, mailer.Invoked, "email should not be sent if reset requested by admin")
+				assert.True(t, tt.user.AdminForcedPasswordReset, "AdminForcedPasswordReset should be true if reset requested by admin")
 			} else {
-				assert.True(st, mailer.Invoked, "email should be sent if vc is not admin")
+				assert.True(t, mailer.Invoked, "email should be sent if vc is not admin")
 				if serviceErr == nil {
 					req, err := ds.FindPassswordResetsByUserID(tt.user.ID)
-					assert.Nil(st, err)
-					assert.NotEmpty(st, req, "user should have at least one password reset request")
+					assert.Nil(t, err)
+					assert.NotEmpty(t, req, "user should have at least one password reset request")
 				}
 			}
 		})
@@ -177,32 +176,34 @@ func TestCreateUser(t *testing.T) {
 		},
 	}
 
-	for i, tt := range createUserTests {
-		payload := kolide.UserPayload{
-			Username:                 tt.Username,
-			Password:                 tt.Password,
-			Email:                    tt.Email,
-			Admin:                    tt.Admin,
-			InviteToken:              tt.InviteToken,
-			AdminForcedPasswordReset: tt.NeedsPasswordReset,
-		}
-		user, err := svc.NewUser(ctx, payload)
-		require.Equal(t, tt.wantErr, err, strconv.Itoa(i))
-		if err != nil {
-			// skip rest of the test if error is not nil
-			continue
-		}
+	for _, tt := range createUserTests {
+		t.Run("", func(t *testing.T) {
+			payload := kolide.UserPayload{
+				Username:                 tt.Username,
+				Password:                 tt.Password,
+				Email:                    tt.Email,
+				Admin:                    tt.Admin,
+				InviteToken:              tt.InviteToken,
+				AdminForcedPasswordReset: tt.NeedsPasswordReset,
+			}
+			user, err := svc.NewUser(ctx, payload)
+			require.Equal(t, tt.wantErr, err)
+			if err != nil {
+				// skip rest of the test if error is not nil
+				return
+			}
 
-		assert.NotZero(t, user.ID)
+			assert.NotZero(t, user.ID)
 
-		err = user.ValidatePassword(*tt.Password)
-		assert.Nil(t, err)
+			err = user.ValidatePassword(*tt.Password)
+			assert.Nil(t, err)
 
-		err = user.ValidatePassword("different_password")
-		assert.NotNil(t, err)
+			err = user.ValidatePassword("different_password")
+			assert.NotNil(t, err)
 
-		assert.Equal(t, user.AdminForcedPasswordReset, *tt.NeedsPasswordReset)
-		assert.Equal(t, user.Admin, *tt.Admin)
+			assert.Equal(t, user.AdminForcedPasswordReset, *tt.NeedsPasswordReset)
+			assert.Equal(t, user.Admin, *tt.Admin)
+		})
 
 	}
 }
@@ -261,19 +262,21 @@ func TestChangeUserPassword(t *testing.T) {
 		},
 	}
 
-	for i, tt := range passwordChangeTests {
-		ctx := context.Background()
-		request := &kolide.PasswordResetRequest{
-			CreatedAt: time.Now(),
-			UpdatedAt: time.Now(),
-			ExpiresAt: time.Now().Add(time.Hour * 24),
-			UserID:    1,
-			Token:     "abcd",
-		}
-		_, err := ds.NewPasswordResetRequest(request)
-		assert.Nil(t, err)
+	for _, tt := range passwordChangeTests {
+		t.Run("", func(t *testing.T) {
+			ctx := context.Background()
+			request := &kolide.PasswordResetRequest{
+				CreatedAt: time.Now(),
+				UpdatedAt: time.Now(),
+				ExpiresAt: time.Now().Add(time.Hour * 24),
+				UserID:    1,
+				Token:     "abcd",
+			}
+			_, err := ds.NewPasswordResetRequest(request)
+			assert.Nil(t, err)
 
-		serr := svc.ResetPassword(ctx, tt.token, tt.newPassword)
-		assert.Equal(t, tt.wantErr, serr, strconv.Itoa(i))
+			serr := svc.ResetPassword(ctx, tt.token, tt.newPassword)
+			assert.Equal(t, tt.wantErr, serr)
+		})
 	}
 }

